File: sketches/sketches.model/src/test/java/edu/kit/kastel/lissa/sketches/model/TestIT.java
package edu.kit.kastel.lissa.sketches.model;

import org.junit.jupiter.api.Test;

import edu.kit.kastel.lissa.sketches.model.elements.uml.IUMLAssociation;
import edu.kit.kastel.lissa.sketches.model.elements.uml.IUMLClass;
import edu.kit.kastel.lissa.sketches.model.elements.uml.IUMLThing;
import edu.kit.kastel.lissa.sketches.model.impl.Box;
import edu.kit.kastel.lissa.sketches.model.impl.Relation;

public class TestIT {
	@Test
	public void test() {
		Sketch sketch = this.getSketch();
		System.out.println(sketch);
	}

	private Sketch getSketch() {
		Sketch sketch = new Sketch();
		sketch.addSketchElement(new Box("Component A", 0.8));
		sketch.addSketchElement(new Box("Component B", 0.6));
		sketch.changeInterpretation(sketch.getBoxElements().get(0), IUMLClass.class);

		sketch.addSketchElement(new Relation("Relation 1", 0.5));
		IUMLAssociation assoc = sketch.changeInterpretation(sketch.getRelationElements().get(0), IUMLAssociation.class);
		sketch.getBoxElements(IUMLThing.class).forEach(e -> assoc.addUMLThingToAssociation(e));
		return sketch;
	}
}

File: sketches/model/src/test/kotlin/edu/kit/kastel/lissa/sketches/model/TestIT.kt
package edu.kit.kastel.lissa.sketches.model

import edu.kit.kastel.lissa.sketches.model.elements.class_diagram.IAssociation
import edu.kit.kastel.lissa.sketches.model.elements.class_diagram.IClass
import edu.kit.kastel.lissa.sketches.model.elements.generic.IInterface
import edu.kit.kastel.lissa.sketches.model.impl.BoxImpl
import edu.kit.kastel.lissa.sketches.model.impl.RelationImpl
import org.junit.jupiter.api.Test

internal class TestIT : TestBase() {
    @Test
    fun test() {
        val sketch = Sketch()
        sketch.addSketchElement(BoxImpl("Component A", 0.8))
        sketch.addSketchElement(BoxImpl("Component B", 0.6))
        sketch.changeInterpretation(sketch.getBoxElements()[0], IClass::class)
        sketch.addSketchElement(RelationImpl("Relation 1", 0.5))
        val assoc =
            sketch.changeInterpretation(sketch.getRelationElements()[0], IAssociation::class)

        sketch.getBoxElements(IClass::class).forEach { e: IClass ->
            assoc.addToAssociation(e)
        }
        sketch.getBoxElements(IInterface::class).forEach { e: IInterface ->
            assoc.addToAssociation(e)
        }
        println(sketch)
    }
}
